src/explore-education-statistics-admin/src/pages/users/UserInvitePage.tsx

```diff
@@ -91,71 +96,122 @@ const UserInvitePage = ({
-                selectedRoleId: Yup.string().required(
-                  'Choose role for the user',
-                ),
-              })}
-              onSubmit={handleSubmit}
-              render={_ => {
-                return (
-                  <Form id={formId}>
-                    <FormFieldset
-                      id={`${formId}-userEmailFieldset`}
-                      legend="Provide the email address for the user"
-                      legendSize="m"
-                      hint="The invited user must have a @education.gov.uk email address"
-                    >
-                      <FormFieldTextInput
-                        id={`${formId}-userEmail`}
-                        label="User email"
-                        name="userEmail"
-                      />
-                    </FormFieldset>
-
-                    <FormFieldset
-                      id={`${formId}-selectedRoleIdFieldset`}
-                      legend="Select the role of the user within the service"
-                      legendSize="m"
-                      hint="Most users should be added with the analyst role"
-                    >
+      {model && (
+        <Formik<FormValues>
+          enableReinitialize
+          initialValues={{
+            userEmail: '',
+            selectedRoleId:
+              orderBy(model.roles, role => role.name)?.[0]?.id ?? '',
+          }}
+          validationSchema={Yup.object({
+            userEmail: Yup.string()
+              .required('Provide the users email')
+              .email('Provide a valid email address'),
+            selectedRoleId: Yup.string().required('Choose role for the user'),
+          })}
+          onSubmit={handleSubmit}
+          render={_ => {
+            return (
+              <Form id={formId}>
+                <FormFieldset
+                  id={`${formId}-userEmailFieldset`}
+                  legend="Provide the email address for the user"
+                  legendSize="m"
+                  hint="The invited user must have a @education.gov.uk email address"
+                >
+                  <FormFieldTextInput
+                    id={`${formId}-userEmail`}
+                    label="User email"
+                    name="userEmail"
+                  />
+                </FormFieldset>
+
+                <FormFieldset
+                  id={`${formId}-selectedRoleIdFieldset`}
+                  legend="Role"
+                  legendSize="m"
+                  hint="The users role within the service."
+                >
+                  <FormFieldSelect
+                    id={`${formId}-selectedRoleId`}
+                    label="Role"
+                    name="selectedRoleId"
+                    options={model?.roles.map(role => ({
+                      label: role.name,
+                      value: role.id,
+                    }))}
+                  />
+                </FormFieldset>
+
+                {/* <FormFieldset
```

Comment:
Why is this all commented out?

Reply:
Part of  EES-622 which i was doing at the same time however they diverged a bit.